components/App.js

```diff
@@ -0,0 +1,105 @@
+import { request } from "../utils/api.js";
+import { initRouter } from "../utils/router.js";
+import NotionEditPage from "./NotionEditPage.js";
+import NotionPage from "./NotionPage.js";
+
+export default function App({ $target }) {
+  // 컨테이너 생성
+  const $notionPageContainer = document.createElement("div");
+  const $notionEditPageContainer = document.createElement("div");
+  $notionPageContainer.className = "notionPageContainer";
+  $notionEditPageContainer.className = "notionEditPageContainer";
+
+  // 페이지 추가 버튼 생성
+  const $addNewRootNotion = document.createElement("button");
+  $addNewRootNotion.className = "addNewRootNotion";
+  $addNewRootNotion.textContent = "+ 페이지 추가";
+
+  // 노션 페이지 컨테이너 토글 버튼 생성
+  const $toggleBtn = document.createElement("button");
+  $toggleBtn.className = "toggle-btn1";
+  $toggleBtn.innerHTML = `<i class="fa fa-solid fa-arrow-right" style="color: #000000;"></i>`;
+
+  $target.appendChild($toggleBtn);
+  $target.appendChild($notionPageContainer);
+  $target.appendChild($notionEditPageContainer);
+  $notionPageContainer.appendChild($addNewRootNotion);
+
+  // 노션 페이지 컨포넌트
+  const notionPage = new NotionPage({
+    $target: $notionPageContainer,
+    initialState: [],
+    onClick: (id) => {
+      const { pathname } = window.location;
+      const [, , notionId] = pathname.split("/");
+      if (id !== notionId) {
+        history.pushState(null, null, `/documents/${id}`);
+        this.route();
+      }
+    },
+    onAdd: (id) => {
+      fetchAddNotion(id);
+    },
+    onDelete: async (id) => {
+      await request(`/documents/${id}`, {
+        method: "DELETE",
+      });
+      history.replaceState(null, null, "/");
+      this.route();
+    },
+    $editorPage: $notionEditPageContainer,
+    $toggleBtn,
+  });
+
+  // 노션 에디터 페이지 컴포넌트
+  const editorPage = new NotionEditPage({
+    $target: $notionEditPageContainer,
+    initialState: "",
```

Comment:
p2;
다른 코드를 살펴보니 model이 아래와 같은 것 같은데요.
객체의 빈상태를 다른 데이터타입이 아닌(문자열) 
- null로 명시적으로 표기하거나, 
- 객체의 속성에 비어있는 문자열로 표시하는 것을 권장드립니다.

```suggestion
    initialState: null,
```

```suggestion
    initialState: { notionId: "", notion: null }
```